### PR TITLE
i18n(zh-cn): Translate `Endpoints.md`

### PR DESCRIPTION
**Pull request by mogeko**
#### What kind of changes does this PR include?

- New or updated content
- Translated content
- Changes to the docs site code

#### Description

Translated [Endpoints document](https://docs.astro.build/zh-cn/core-concepts/endpoints/). Its English reference comes from [here](https://docs.astro.build/en/core-concepts/endpoints/).

At the same time, we modified a code error for [`/en/core-concepts/endpoints.md`](https://github.com/withastro/docs/blob/6d387a144de27ee78b75a740623489907f174163/src/pages/en/core-concepts/endpoints.md?plain=1#L32-L38):

https://github.com/withastro/docs/blob/6d387a144de27ee78b75a740623489907f174163/src/pages/en/core-concepts/endpoints.md?plain=1#L32-L38

```ts
export async function get({ params, request }) { 
   const response = await fetch("https://astro.build/assets/press/full-logo-light.png"); 
   const buffer = Buffer.from(await response.arrayBuffer()); 
   return { 
     body: buffer, 
     encoding: 'binary', 
   };
}
```

He was marked as `missing` in #438.

Please labeled this PR with `hacktoberfest-accepted`. :kissing_heart:

### CODE REVIEW

**Review by zihan-ch — CHANGES_REQUESTED** [review]
Change the nav.ts too. It's at `src/i18n/zh-cn/nav.ts`.  Add this line:
```
'core-concepts/endpoints': 'API 端点',
```

File: src/pages/zh-cn/core-concepts/endpoints.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+在静态生成的站点中，你的自定义端点在构建时被调用以生成静态文件。如果你选择启用 [SSR](/zh-cn/guides/server-side-rendering/) 模式，自定义端点会变成根据请求调用的实时服务器端点。静态和 SSR 端点的定义类似，但 SSR 端点支持附加额外的功能。
+
+## 静态文件端点
+
+要想要创建自定义端点，请将 `.js` 或 `.ts` 文件添加到 `/pages` 目录。`.js` 或 `.ts` 这样的扩展名将在构建过程中被删除，因此文件名中应包含你要创建的数据的扩展名。例如，文件 `src/pages/data.json.ts` 将被构建为 API 端点 `/data.json`。
+
+端点导出一个 `get` 函数（可选的以 `async` 导出），它的唯一参数是一个对象，其具有两个属性（`params` 和 `request`）。它返回一个带有 `body` 的对象，Astro 将在构建时调用它并使用 `body` 中的内容来生成文件。
+
+```js
+// Example: src/pages/builtwith.json.ts
```
Comment (zihan-ch) [inline src/pages/zh-cn/core-concepts/endpoints.md:18]:
```suggestion
// 例子: src/pages/builtwith.json.ts
```

File: src/pages/zh-cn/core-concepts/endpoints.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+// Outputs: /builtwith.json
```
Comment (zihan-ch) [inline src/pages/zh-cn/core-concepts/endpoints.md:19]:
```suggestion
// 输出: /builtwith.json
```

File: src/pages/zh-cn/core-concepts/endpoints.md [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+  return {
+    body: JSON.stringify({
+      name: 'Astro',
+      url: 'https://astro.build/',
+    }),
+  };
+}
+```
+
+返回对象还可以具有 `encoding` 属性。它可以是任何可以被 Node.js 的 `fs.writeFile` 方法接收的 [`BufferEncoding`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bdd02508ddb5eebcf701fdb8ffd6e84eabf47885/types/node/buffer.d.ts#L169)。例如，想要生成一个二进制的 PNG 图像：
```
Comment (zihan-ch) [inline src/pages/zh-cn/core-concepts/endpoints.md:30]:
```suggestion
返回对象还可以具有 `encoding` 属性。它可以是任何可以被 Node.js 的 `fs.writeFile` 方法接收的 [`BufferEncoding`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bdd02508ddb5eebcf701fdb8ffd6e84eabf47885/types/node/buffer.d.ts#L169)。例如，生成一个二进制的 PNG 图像：
```

File: src/pages/zh-cn/core-concepts/endpoints.md [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+
+```ts title="src/pages/astro-logo.png.ts" {6}
+export async function get({ params, request }) => {
+  const response = await fetch("https://astro.build/assets/press/full-logo-light.png");
+  const buffer = Buffer.from(await response.arrayBuffer());
+  return {
+    body: buffer,
+    encoding: 'binary',
+  };
```
Comment (zihan-ch) [inline src/pages/zh-cn/core-concepts/endpoints.md:39]:
```suggestion
  };
}
```

File: src/pages/zh-cn/core-concepts/endpoints.md [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+```js
+import type { APIRoute } from 'astro';
+
+export const get: APIRoute = async function get ({params, request}) {
+...
+```
+
+### `params` 和动态路由
+
+端点同页面一样，都支持 Astro 的[动态路由](/zh-cn/core-concepts/routing/#动态路由)功能。使用中括号包裹参数作为文件名，并导出 [`getStaticPaths()` 函数](/zh-cn/reference/api-reference/#getstaticpaths)。然后，你可以使用传递给端点函数（`get`）的 `params` 属性访问参数：
```
Comment (zihan-ch) [inline src/pages/zh-cn/core-concepts/endpoints.md:53]:
```suggestion
API 端点同页面一样，都支持 Astro 的[动态路由](/zh-cn/core-concepts/routing/#动态路由)功能。使用中括号包裹参数作为文件名，并导出 [`getStaticPaths()` 函数](/zh-cn/reference/api-reference/#getstaticpaths)。然后，你可以使用传递给 API 端点函数（`get`）的 `params` 属性访问参数：
```

File: src/pages/zh-cn/core-concepts/endpoints.md [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+
+```ts title="src/pages/[id].json.ts"
+import type { APIRoute } from 'astro';
+
+const usernames = ["Sarah", "Chris", "Dan"]
```
Comment (zihan-ch) [inline src/pages/zh-cn/core-concepts/endpoints.md:58]:
```suggestion
const usernames = ["张三", "李四", "王五"]
```

File: src/pages/zh-cn/core-concepts/endpoints.md [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+        { params: { id: "2"} },
+    ]
+}
+```
+
+这将在构建时生成三个 JSON 端点：`/api/1.json`、`/api/2.json` 和 `/api/3.json`。带端点的动态路由与页面的工作方式相同，但由于端点是一个函数而不是组件，因此 [props](/zh-cn/reference/api-reference/#通过-props-传递数据) 是不被支持的。
+
+### `request`
+
+就像[上面提到的](#静态文件端点)，端点函数的唯一参数是一个对象，此对象包含一个名为 `request` 的属性；但在静态模式下，你只能访问 `request.url`。这个属性将返回当前端点的完整 URL，并且与 [Astro.request.url](/zh-cn/reference/api-reference/#astrorequest) 对页面的作用相同。
```
Comment (zihan-ch) [inline src/pages/zh-cn/core-concepts/endpoints.md:82]:
```suggestion
就像[上面提到的](#静态文件端点)，端点函数的唯一参数是一个对象，此对象包含一个名为 `request` 的属性；但在静态模式下，你只能使用 `request.url`。这个属性将返回当前端点的完整 URL，并且与 [Astro.request.url](/zh-cn/reference/api-reference/#astrorequest) 对页面的作用相同。
```

File: src/pages/zh-cn/core-concepts/endpoints.md [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+    body: JSON.stringify({
+      message: "This was a GET!"
+    })
+  }
+};
+
+export const post: APIRoute = ({ request }) => {
+  return {
+    body: JSON.stringify({
+      message: "This was a POST!"
```
Comment (zihan-ch) [inline src/pages/zh-cn/core-concepts/endpoints.md:155]:
```suggestion
      message: "这是一个 POST 请求！"
```

File: src/pages/zh-cn/core-concepts/endpoints.md [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+    })
+  }
+}
+
+export const del: APIRoute = ({ request }) => {
+  return {
+    body: JSON.stringify({
+      message: "This was a DELETE!"
```
Comment (zihan-ch) [inline src/pages/zh-cn/core-concepts/endpoints.md:163]:
```suggestion
      message: "这是一个 DELETE 请求！"
```

File: src/pages/zh-cn/core-concepts/endpoints.md [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+    })
+  }
+}
+
+export const all: APIRoute = ({ request }) => {
+  return {
+    body: JSON.stringify({
+      message: `This was a ${request.method}!`
```
Comment (zihan-ch) [inline src/pages/zh-cn/core-concepts/endpoints.md:171]:
```suggestion
      message: `这是一个 ${request.method} 请求！`
```

File: src/pages/zh-cn/core-concepts/endpoints.md [hunk 11]
```diff
@@ -0,0 +1,279 @@
...
+
+在 SSR 模式下，`request` 属性返回一个可用的 [`Request`](https://developer.mozilla.org/zh-CN/docs/Web/API/Request) 对象，该对象引用当前请求。这允许你接收数据并检查 [`headers`](https://developer.mozilla.org/zh-CN/docs/Web/API/Request/headers)：
+
+```ts title="src/pages/test-post.json.ts"
+export const post: APIRoute = async ({ request }) => {
+  if (request.headers.get("Content-Type") === "application/json") {
+    const body = await request.json();
+    const name = body.name;
+    return new Response(JSON.stringify({
+      message: "Your name was: " + name
```
Comment (zihan-ch) [inline src/pages/zh-cn/core-concepts/endpoints.md:187]:
```suggestion
      message: "你的名字是：" + name
```

**Review by zihan-ch — CHANGES_REQUESTED** [review]

File: src/pages/zh-cn/core-concepts/endpoints.md (hunk 4 above)
Comment (mogeko) [reply src/pages/zh-cn/core-concepts/endpoints.md:39]: It seems that this error also appears in the [English version](https://github.com/withastro/docs/blob/6d387a144de27ee78b75a740623489907f174163/src/pages/en/core-concepts/endpoints.md?plain=1#L38).

Comment (zihan-ch) [reply src/pages/zh-cn/core-concepts/endpoints.md:39]: > It seems that this error also appears in the [English version](https://github.com/withastro/docs/blob/6d387a144de27ee78b75a740623489907f174163/src/pages/en/core-concepts/endpoints.md?plain=1#L38).

Yes. I think we can change it in this PR.

**Review by delucis — APPROVED** [review]
Looks great! Thank you for the awesome team work on this one @mogeko & @zihan-ch! 💜

**Pull request merged**